app/task_routes.py

```diff
@@ -0,0 +1,140 @@
+    task_list = []
+    
+    """HELPER FUNCTION TO DETERMINE IF TASK IS COMPLETED"""
+    def is_complete():
+        if "completed_at" in task_list == None:
+            return True
+        else:
+            return False
+
+    task_response = []
+    for task in tasks:
+        task_response.append({
+        "id":task.task_id,
+        "title":task.title,
+        "description":task.description,
+        "is_complete":is_complete()
+            }) 
+
+    if sort_request == "asc":
+        task_response = sorted(task_response, key=lambda a: a["title"])
+    elif sort_request == "desc":
+        task_response = sorted(task_response, key=lambda d: d["title"], reverse=True) 
+
+    return jsonify(task_response)
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+    if task.task_id:
+        return {"task":task.to_dict()}
+    else:
+        return {"message": f"Task {task_id} not found"}, 404
+
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+    if task:
+        task_dict = {
+        "details": f"Task {task_id} \"{task.title}\" successfully deleted"
+        }
+    else:
+        return jsonify({"message": f"Task {task_id} not found"}), 404
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return jsonify(task_dict), 200
+    
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+    if task:   
+        task.title = request_body["title"]
+        task.description = request_body["description"]
+        response_body = {"task": {
+                    "id": 1,
+                    "title": "Updated Task Title",
+                    "description": "Updated Test Description",
+                    "is_complete": False
+            }}
+        db.session.commit()
+        return jsonify(response_body), 200
+    else:
+        db.session.commit()
+        return jsonify({"message": f"Task {task_id} not found"}), 404
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+    task = Task.query.get_or_404(task_id)
+    task.completed_at = datetime.now()
+    db.session.commit()
+    return jsonify(task=task.to_dict()), 200
+
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(task_id):
+    task = Task.query.get_or_404(task_id)
+    task.completed_at = None
+    db.session.commit()
+    return jsonify(task=task.to_dict()), 200
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete_on_completed_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    # task_completed_at = datetime.now()
+
+    db.session.commit()
+    
+    return jsonify(task=task.to_dict()), 200
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
```

Comment:
I think you meant to register this endpoint as `/<task_id>/mark_incomplete`

However, see my comment above about Flask not being to determine which method should execute if the route is exactly the same. 

Your mark_incomplete method should be able to handle marking incomplete on a complete task and marking incomplete on an incomplete task